hpeOneView/resources/servers/id_pools_ipv4_ranges.py

```diff
@@ -24,106 +24,72 @@
     """
     The ID pools IPv4 ranges resource provides a Client API for managing IPv4 ranges.
     """
     URI = '/rest/id-pools/ipv4/ranges'
 
-    def __init__(self, con):
-        self._client = ResourceClient(con, self.URI)
+    def __init__(self, connection, data=None):
+        super(IdPoolsIpv4Ranges, self).__init__(connection, data)
 
-    def create(self, resource, timeout=-1):
-        """
-        Creates an IPv4 range.
-
-        Args:
-            resource (dict): Object to create
-            timeout: Timeout in seconds. Wait for task completion by default. The timeout does not abort the operation
-                in OneView; it just stops waiting for its completion.
-
-        Returns:
-            dict: Created range.
-        """
-        return self._client.create(resource, timeout=timeout)
-
-    def get(self, id_or_uri):
-        """
-        Gets an IPv4 range by ID or URI.
-
-        Using the allocator and collector associated with the range, IDs may be allocated from or collected back to the
-        range.
-
-        Args:
-            id_or_uri: Can be either the range ID or URI.
-
-        Returns:
-            dict: Range
-        """
-        return self._client.get(id_or_uri)
-
-    def enable(self, information, id_or_uri, timeout=-1):
+    def enable(self, information, uri, timeout=-1):
         """
         Enables or disables an IPv4 range.
 
         Args:
             information (dict): Information to update.
-            id_or_uri: ID or URI of range.
+            uri: URI of range.
             timeout: Timeout in seconds. Wait for task completion by default. The timeout does not abort the operation
                 in OneView; it just stops waiting for its completion.
 
         Returns:
             dict: Updated IPv4 range.
         """
+        return self._helper.update(information, uri, timeout=timeout)
 
-        uri = self._client.build_uri(id_or_uri)
-
-        return self._client.update(information, uri, timeout=timeout)
-
-    def update(self, information, timeout=-1):
+    def allocator(self, information, uri, timeout=-1):
         """
-        Edit an IPv4 Range.
+        Allocates a set of IDs from an IPv4 range.
 
         Args:
             information (dict): Information to update.
+            id_or_uri: URI of range.
             timeout: Timeout in seconds. Wait for task completion by default. The timeout does not abort the operation
                 in OneView; it just stops waiting for its completion.
 
         Returns:
-            dict: Updated IPv4 range.
+            dict: The allocator returned contains the list of IDs successfully allocated.
         """
+        uri = uri + "/allocator"
+        return self._helper.update(information, uri, timeout=timeout)
 
-        return self._client.update(information, timeout=timeout)
-
-    def delete(self, resource, force=False, timeout=-1):
+    def collector(self, information, uri, timeout=-1):
```

Comment:
method will be easy to understand if it is "update_collector"

Reply:
done